src/Build/BackEnd/Components/FileAccesses/IFileAccessManager.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Threading;
+using Microsoft.Build.BackEnd;
+using Microsoft.Build.Framework.FileAccess;
+
+namespace Microsoft.Build.FileAccesses
+{
+    internal interface IFileAccessManager
+    {
+        void ReportFileAccess(FileAccessData fileAccessData, int nodeId);
+
+        void ReportProcess(ProcessData processData, int nodeId);
+
+        // Note: HandlerRegistration is exposed directly instead of IDisposable to avoid boxing.
```

Comment:
I'll clarify the comment, but it means `FileAccessManager.HandlerRegistration`, ie the return type. It's a struct which implements `IDisposable` and so instead of the return type being `IDisposable`, it's the struct type so that it doesn't get boxed.

Tbh, this is called exactly once per plugin, of which there will likely be exactly 1 in most cases, if any. So this doesn't actually save much =P